### ISSUE TITLE
[receiver/solace] Tests failing on Windows

### ISSUE THREAD

**Issue by dmitryax** [open-telemetry/opentelemetry-collector-contrib#17197]
The following tests are failing on Windows after https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16570:

```
--- FAIL: TestReceiverFlowControlDelayedRetry (0.04s)
    --- FAIL: TestReceiverFlowControlDelayedRetry/Without_error (0.02s)
    --- PASS: TestReceiverFlowControlDelayedRetry/With_error (0.01s)
=== RUN   TestReceiverFlowControlDelayedRetryInterrupt
--- PASS: TestReceiverFlowControlDelayedRetryInterrupt (0.02s)
=== RUN   TestReceiverFlowControlDelayedRetryMultipleRetries
    receiver_test.go:507: 
        	Error Trace:	D:\a\opentelemetry-collector-contrib\opentelemetry-collector-contrib\receiver\solacereceiver\receiver_test.go:507
        	Error:      	receiveMessage did not return after some time
        	Test:       	TestReceiverFlowControlDelayedRetryMultipleRetries
--- FAIL: TestReceiverFlowControlDelayedRetryMultipleRetries (0.04s)
```

See https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3754566082/jobs/6378899100

It was not caught before because of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/17181

**Comment by dmitryax** [comment]
@mcardy looks like it's still failing on main. Please see https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3760209755/jobs/6390667040

**Comment by mcardy** [comment]
Wow looks like its very slow. I wonder why windows is so slow. I ran it about 100 times on my own windows machine and it seemed fine with the fix. I'll open a PR to up the timeout even more/investigate alternatives.

**Comment by dmitryax** [comment]
Fixed by https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17204

**Comment by dmitryax** [comment]
@mcardy It's still happening occasionally https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3976471954/jobs/6816975722

Maybe we should increase the timeout for windows to 1 second?

**Comment by codeboten** [comment]
Happened again [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4117979899/jobs/7109942768), re-opening

**Comment by mcardy** [comment]
Looking at this again, this is a different test failing. This is addressed with #19409
